src/Cli.Tests/ValidateConfigTests.cs

```diff
@@ -114,6 +114,37 @@ public void TestValidateWithEmptyConfig()
         }
     }
 
+    /// <summary>
+    /// This Test is used to verify that the validate command is able to catch invalid values for the depth-limit property.
+    /// </summary>
+    [DataTestMethod]
+    [DataRow("null", true, false, DisplayName = "Invalid Value: 'null'")]
+    [DataRow("20", true, true, DisplayName = "Valid Value: 20")]
+    [DataRow(-1, false, false, DisplayName = "Invalid Value: -1")]
+    [DataRow(0, false, false, DisplayName = "Invalid Value: 0")]
+    [DataRow(-2, false, false, DisplayName = "Invalid Value: -2")]
+    [DataRow(2147483648, false, false, DisplayName = "Invalid Value: 2147483648")]
```

Comment:
Display names would be more description if you listed why the value is invalid. isn't `-1` valid to denote "no depth limit".

Unclear why the string value `"20"` is a valid input for query-depth, that isn't an integer. That test case also contradicts the test method name which ensures invalid values result in failure

Reply:
Thanks for catching that. Actually the parameter `isSuccess` is not being used and we are only checking for failure. that's why both "20" as well as "-1" failed. "-1" failed because of invalid connection string as these are unit tests so we don't have valid connection string in CLI tests. Once the engine work is done we can have tests in engine which will validate the complete flow.